ui/src/datasource/DatasourceOverview.vue

```diff
@@ -154,29 +175,64 @@ export default class DatsourceOverview extends Vue {
   }
 
   private onCreatePipeline (datasource: Datasource): void {
-    this.$router.push({ name: 'pipeline-new', params: { datasourceId: `${datasource.id}` } })
+    this.$router
+      .push({
+        name: 'pipeline-new',
+        params: { datasourceId: `${datasource.id}` },
+      })
       .catch(error => console.log('Failed to route to pipeline-new', error))
   }
 
   private filterOnlyDisplayName (value: unknown, search: string, item: Datasource): boolean {
-    return value != null &&
-          search != null &&
-          typeof value === 'string' &&
-          item.metadata.displayName.toLocaleLowerCase().includes(search.toLocaleLowerCase())
-  }
-
-  private getHoursFromMS (intervalInMS: number): number {
-    return Math.floor(intervalInMS / ONE_HOUR_IN_MS)
-  }
-
-  private getMinutesFromMS (intervalInMS: number): number {
-    return Math.floor((intervalInMS % ONE_HOUR_IN_MS) / ONE_MINUTE_IN_MS)
+    return (
+      value != null &&
+      search != null &&
+      typeof value === 'string' &&
+      item.metadata.displayName.toLocaleLowerCase().includes(search.toLocaleLowerCase())
+    )
   }
 
   private async loadDataSources (): Promise<void> {
     this.isLoadingDatasources = true
     this.datasources = await DatasourceREST.getAllDatasources()
     this.isLoadingDatasources = false
   }
+
+  private datasourceToJson (datasource: Datasource): string {
+    return JSON.stringify(datasource, null, 2)
+  }
+
+  private trimQueryParams (url: string): string {
+    const queryParamRegex = /\?.*/
+
+    const urlContainsQueryParams = queryParamRegex.test(url)
+
+    if (urlContainsQueryParams) {
+      return url.replace(queryParamRegex, '') + '...'
+    }
+    
+    return url
+  }
 }
 </script>
+<style>
+td {
+  word-break: break-all !important;
+}
+
+.editor {
+  width: 100%;
+  height: 500px;
+}
```

Comment:
I wonder if there is a way to make the editor always the required height without specifying a pixel value.

As it is right now, "normal" configs fit in perfectly (on my screen/machine at least), configs with a longer URL produce a scroll bar for 1 line.

Reply:
I also dislike the fixed height. I did some research on the monaco editor regarding this without much success but i can look into it further.

Reply:
I'm wondering if the editor was an overkill here. Since no editing will be done, the high configuration cost probably isn't justified. I think it's enough to simply wrap the JSON in `<pre>...</pre>`. A bit sad that basic syntax highlighting isn't build into HTML.

![image](https://user-images.githubusercontent.com/29175032/102460073-c2802b00-4046-11eb-9c20-ce81002dbb1e.png)

Reply:
I wondered about that, too, and I think i prefer the simple version with the `<pre>`-tag.